dotnet/src/Agents/UnitTests/Yaml/AzureAIKernelAgentYamlTests.cs

```diff
@@ -0,0 +1,346 @@
+                  authentication:
+                      audience: audience
+            """;
+        AzureAIAgentFactory factory = new();
+        this.SetupResponse(HttpStatusCode.OK, AzureAIAgentFactoryTests.AzureAIAgentResponse);
+
+        // Act
+        var agent = await factory.CreateAgentFromYamlAsync(text, this._kernel);
+
+        // Assert
+        Assert.NotNull(agent);
+        var requestContent = Encoding.UTF8.GetString(this._messageHandlerStub.RequestContent!);
+        Assert.NotNull(requestContent);
+        var requestJson = JsonSerializer.Deserialize<JsonElement>(requestContent);
+        Assert.Equal(3, requestJson.GetProperty("tools").GetArrayLength());
+        Assert.Equal("openapi", requestJson.GetProperty("tools")[0].GetProperty("type").GetString());
+        Assert.Equal("openapi", requestJson.GetProperty("tools")[1].GetProperty("type").GetString());
+        Assert.Equal("openapi", requestJson.GetProperty("tools")[2].GetProperty("type").GetString());
+    }
+
+    /// <summary>
+    /// Verify the request includes a Sharepoint tool when creating an Azure AI agent.
+    /// </summary>
+    [Fact]
+    public async Task VerifyRequestIncludesSharepointGroundingAsync()
+    {
+        // Arrange
+        var text =
+            """
+            type: azureai_agent
+            name: AzureAIAgent
+            description: AzureAIAgent Description
+            instructions: AzureAIAgent Instructions
+            model:
+              id: gpt-4o-mini
+            tools:
+                - type: sharepoint_grounding
+                  tool_connections:
+                    - connection_string
+            """;
+        AzureAIAgentFactory factory = new();
+        this.SetupResponse(HttpStatusCode.OK, AzureAIAgentFactoryTests.AzureAIAgentResponse);
+
+        // Act
+        var agent = await factory.CreateAgentFromYamlAsync(text, this._kernel);
+
+        // Assert
+        Assert.NotNull(agent);
+        var requestContent = Encoding.UTF8.GetString(this._messageHandlerStub.RequestContent!);
+        Assert.NotNull(requestContent);
+        var requestJson = JsonSerializer.Deserialize<JsonElement>(requestContent);
+        Assert.Equal(1, requestJson.GetProperty("tools").GetArrayLength());
+        Assert.Equal("sharepoint_grounding", requestJson.GetProperty("tools")[0].GetProperty("type").GetString());
+    }
+
+    #region private
+    private void SetupResponse(HttpStatusCode statusCode, string response) =>
```

Comment:
Return the HttpResponse and let the caller `using` it. Will prevent the `pragma warning`.

```suggestion
    private HttpResponse SetupResponse(HttpStatusCode statusCode, string response) =>
```

Reply:
Using `ResponseToReturn` instead